app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -44,16 +44,63 @@ def construct_name(family)
       end
 
       def medium_uri_with_content_uri(host, url = nil)
-        if host.try(:content_source) && (repo_details = kickstart_repo(host))
-          URI.parse(repo_details[:path])
+        if host.try(:content_source) && (repos = distribution_repositories(host)).present?
+          if repos.size == 1
+            # There is exactly 1 ks repo distribution matching the os, arch, release
+            # cv, env for this host. Just get the media path from available repo.
+            URI.parse(repos.first.full_path(host.content_source))
+          else
+            # There is more then one possible  kickstart repository distribution
+            # for this host. This means the user must have chosen an install media as
+            # the repo path for install media content.
+            # This will typically look like ->
+            # path: http://<sat-fqdn>/pulp/repos/Default_Organization/Library/content/dist/rhel/workstation/7/7.2/x86_64/kickstart/
+            # what we need though is the sat-fqdn up replaced by the content source/ capsule/smartproxy  url.
+            #
+            # So we first get the Install Media path
+            # i.e http://<sat-fqdn>/pulp/repos/Default_Organization/Library/content/dist/rhel/workstation/7/7.2/x86_64/kickstart/
+            media_uri = medium_uri_without_content_uri(host, url)
```

Comment:
We need to be sure that the selected media path is actually part of one of the kickstart repos. 

So for example if the user selects content source and the os has media associated to it that is not part of a ks repo we need to handle that. We just need to be sure.